src/ngMessages/messages.js

```diff
@@ -409,8 +437,15 @@ angular.module('ngMessages', [], function initAngularHelpers() {
           });
 
           if (unmatchedMessages.length !== totalMessages) {
+            // Unset default message if set
+            if (ctrl.default) ctrl.default.detach();
+
             $animate.setClass($element, ACTIVE_CLASS, INACTIVE_CLASS);
           } else {
+
+            // Set default message if keys in collection do not match any message
+            if (ctrl.default && truthyKeys > 0) ctrl.default.attach();
+
             $animate.setClass($element, INACTIVE_CLASS, ACTIVE_CLASS);
```

Comment:
This line should only be called if there is no default...  Or perhaps this whole block should look like:

```js
var messageMatched = unmatchedMessages.length !== totalMessages;

if (ctrl.default) {
  if (messageMatched) {
    ctrl.default.detach();
  } else if (truthyKeys > 0) {
    ctrl.default.attach();
  }
}
if (messageMatched || ctrl.default) {
  $animate.setClass($element, ACTIVE_CLASS, INACTIVE_CLASS);
} else {
  $animate.setClass($element, INACTIVE_CLASS, ACTIVE_CLASS);
}
```